packages/react-scripts/config/webpack.config.dev.js

```diff
@@ -199,16 +199,6 @@ module.exports = {
         ],
         include: paths.appSrc,
       },
-      {
-        // `mjs` support is still in its infancy in the ecosystem, so we don't
-        // support it.
-        // Modules who define their `browser` or `module` key as `mjs` force
-        // the use of this extension, so we need to tell webpack to fall back
-        // to auto mode (ES Module interop, allows ESM to import CommonJS).
-        test: /\.mjs$/,
-        include: /node_modules/,
-        type: 'javascript/auto',
-      },
       {
```

Comment:
☝️ Using `javascript/auto` may make `.mjs` easier to work with. I believe letting it default to `javascript/esm` makes things a bit more strict _(prevents some stub/mocks from working with bundled code)_. Might be worth seeing if it can continue to be type `javascript/auto`.

Reply:
Thank you for the insight. We should probably apply this to *all* files then and not just `node_modules`.
Does that sound good?

Reply:
Would it be fair to say we don't want the `javascript/esm` mode until we can prevent ESM from being used in `.js` files?

Reply:
@jdalton How ready is `esm` to be used as a transform in Jest? Do you think this would be a good default while we still support users using ESM in `.js` files?

We'd really like to achieve maximum interoperability while not breaking the behavior of the specification, and I dunno how spec-aligned `esm` (the package) is.

Reply:
/cc @jaydenseric

Reply:
`.mjs` intentionally sends a very strong signal that the mode is to be `esm`, and for how CJS may be imported. No one should be putting anything other than correct ESM in an `.mjs` file, and it is a grave mistake to facilitate or encourage misuse.

It is far preferable that a bundling error throws so people can fix (and learn about) their flawed `.mjs` or stop using it, rather than mask issues and then the `.mjs` won't run natively in Node.js or in other tools that do fairly assume that `.mjs` means ESM.

Reply:
Would this imply we should leave on webpack's handling of `.mjs` files as `javascript/esm`?